crates/boojum/src/gadgets/non_native_field/implementations/implementation_u16.rs

```diff
@@ -136,22 +136,7 @@ where
         // well, we just mul by 1
         let mut one: NonNativeFieldOverU16<F, T, N> =
             Self::allocated_constant(cs, T::one(), &self.params);
-        let mut normalized = self.mul(cs, &mut one);
-
-        // assert that we only have "modulus limbs" moduluses in this element
-        assert_eq!(normalized.non_zero_limbs, normalized.params.modulus_limbs);
-
-        // sub modulus
-        let modulus = self
-            .params
-            .modulus
-            .map(|el| cs.allocate_constant(F::from_u64_unchecked(el as u64)));
```

Comment:
I'm not an expert, but since this (seemingly) changes the layout of the circuit, is it a breaking change (e.g. it requires to generate new keys)?

Reply:
Yes, a new key needs to be generated. However, this is mostly a performance improvement and unrelated to security, so it is not urgent to fix.

Reply:
I've marked it as a breaking change though: note the `!` in the `fix!` (it instructs the release bot that the change is breaking), and applied `Protocol upgrade` label (for general awareness -- so that this PR is not merged accidentally).
Please do so in the future PRs if you know that it requires new VKs.

Also cc @mm-zk --- IIRC you maintain a table of tasks/PRs that should be included for each upgrade -- please add this PR there.